libnmstate/nm/applier.py

```diff
@@ -377,9 +382,17 @@ def _build_connection_profile(iface_desired_state, base_con_profile=None):
     con_setting.set_master(master, master_type)
     settings.append(con_setting.setting)
 
-    wired_setting = wired.create_setting(iface_desired_state, base_profile)
-    if wired_setting:
-        settings.append(wired_setting)
+    # Don't create wired/ethernet setting when not explicitly asked in
```

Comment:
This is what I understand:
- If there is an attribute, then we will copy the existing ones from `base_profile` and add the mentioned attributes.
- If there is nothing mentioned in the desired state by the client, the wired setting will not be added to the new profile, therefore anything that existed will get removed.

Reply:
Actually, the `Update2()` will not remove setting if you don't defined it.

But the `nm/applier.py` does not need to know that detail.
So when desire state does not have wired config, pass the `{}` to `create_setting()`.